### DISCUSSION TITLE
A middleware that intercepts ip

### DISCUSSION THREAD

**Discussion by laazua** [fastapi/fastapi#6274] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typing
from starlette.types import ASGIApp
from fastapi import Request, Response
from starlette.middleware.base import BaseHTTPMiddleware, RequestResponseEndpoint, DispatchFunction


class AllowIps(BaseHTTPMiddleware):
    def __init__(self, app: ASGIApp, ips: typing.Optional[list] = None, dispatch: typing.Optional[DispatchFunction] = None) -> None:
        super().__init__(app, dispatch)
        self.ips = ips

    async def dispatch(self, request: Request, call_next: RequestResponseEndpoint) -> Response:
        response = await call_next(request)
        if request.client.host not in self.ips:
            print("xxxxxxxxxxx")
            return {"msg": "ip not allowed"}
        return response
```

### Description

* I have a requirement that the unsatisfied ip does not allow access to my application.
* The custom middleware seems to have to return a call_next object, how can I do this.

### Operating System

Linux

### Operating System Details

centos 7

### FastAPI Version

0.79.0

### Python Version

3.9

### Additional Context

  File "/home/gamecpp/.local/share/virtualenvs/mgsev-tkZT_j_i/lib/python3.9/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/gamecpp/.local/share/virtualenvs/mgsev-tkZT_j_i/lib/python3.9/site-packages/starlette/middleware/base.py", line 69, in __call__
    await response(scope, receive, send)
TypeError: 'dict' object is not callable

**Comment by laazua** [comment]
i solved it

**Comment by JarroVGIT** [comment]
IT would be great if you would share how you solved it, for people running into the same problem and looking for an answer like you did :)

**Comment by laazua** [comment, marked as answer]
> IT would be great if you would share how you solved it, for people running into the same problem and looking for an answer like you did :)

just return Response object, and it would be solved. like this:
async def dispatch(self, request: Request, call_next: RequestResponseEndpoint) -> Response:
            if request.client.host not in self.ips:
                    return Response("your ip not allowed")
            else:
                   return await call_next(request)
